test/model_creation_tests.jl

```diff
@@ -35,6 +35,13 @@ using Test, Agents, Random
     end
     @test Fisher <: AbstractHuman
     @test :fish_per_day ∈ fieldnames(Fisher)
+
```

Comment:
Here definitely more tests should have been added to see how this works with `add_agent!` that automatically creates agents.

Reply:
yes, it is actually something I didn't consider when doing the pr :D, it seems that it shouldn't have any effect on the version with `args`, I'm also pretty sure that currently if we pass anything in `kwargs` it will throw error (even before this pr)

Reply:
But we can support both ways with:

```julia
function add_agent!(
    pos::ValidPos,
    A::Type{<:AbstractAgent},
    model::ABM,
    properties::Vararg{Any, N};
    kwproperties...,
) where {N}
    id = nextid(model)
    if isempty(kwproperties)
        newagent = A(id, pos, properties...)
    else
        newagent = A(; id = id, pos = pos, kwproperties...)
    end
    add_agent_pos!(newagent, model)
end
```

It should be as fast as before since the branch prediction is pretty easy so the compiler should be able able to infer what to do

Reply:
indeed I don't see any meaningful difference in our model comparisons

```julia
# before this modification
Agents.jl WolfSheep-small (ms): 2.8282849999999997
Agents.jl WolfSheep-large (ms): 99.588342
Agents.jl Flocking-small (ms): 9.377578999999999
Agents.jl Flocking-large (ms): 67.284021
Agents.jl Schelling-small (ms): 0.453293
Agents.jl Schelling-large (ms): 4.035073
Agents.jl ForestFire-small (ms): 0.767832
Agents.jl ForestFire-large (ms): 16.152233

# after this modification
Agents.jl WolfSheep-small (ms): 2.82549
Agents.jl WolfSheep-large (ms): 99.25996699999999
Agents.jl Flocking-small (ms): 8.799429
Agents.jl Flocking-large (ms): 67.878292
Agents.jl Schelling-small (ms): 0.455848
Agents.jl Schelling-large (ms): 4.059518
Agents.jl ForestFire-small (ms): 0.776428
Agents.jl ForestFire-large (ms): 16.248959
```